lightning/src/ln/channelmanager.rs

```diff
@@ -1627,7 +1665,11 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 	/// Users need to notify the new ChannelManager when a new block is connected or
 	/// disconnected using its `block_connected` and `block_disconnected` methods, starting
 	/// from after `params.latest_hash`.
-	pub fn new(fee_est: F, chain_monitor: M, tx_broadcaster: T, logger: L, keys_manager: K, config: UserConfig, params: ChainParameters) -> Self {
+	pub fn new_public(fee_est: F, chain_monitor: M, tx_broadcaster: T, router: R, logger: L, keys_manager: K, config: UserConfig, params: ChainParameters) -> Self {
```

Comment:
Hmm... I question whether moving the retry logic into ChannelManager would be more work than initial trampoline. It's not much. Since I almost have the changeset, think we could withhold judgment until a draft PR is opened? If it seems like way more work, sure thing

Reply:
> I wonder if it makes sense to jump to moving the retry logic into ChannelManager directly now (by just moving the code into a new module in `ln` and wiring up the events to flow through the payer instead of getting passed out) now, giving us a good reason for putting the router here _and_ letting users clean up their code all in one go, and then use that for trampoline? Sorry for just raising this now, realize I should have thought this through earlier.

Maybe I'm misunderstanding, but unless we hardcode the `Router` implementation there, we'd still need to parameterize `ChannelManager` given `InvoicePayer` has it's own parameters (e.g., `Router`), no? Of course, hardcoding with `DefaultRouter` also means parameterization for `NetworkGraph` and `Score`...

Reply:
Okay, I'm gonna move forward with moving retries into `ChannelManager` on top of this PR. Likely will want to base it on #1830 as well